### PR TITLE
fix doc for BASE_DIR in add_*_files

### PR DESCRIPTION
**Pull request by vrabaud**
As detailed in cmake/genmsg-extras.cmake.em

### CODE REVIEW

File: doc/usermacros.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@ User macro reference
    :param DIRECTORY: Directory containing messages.  May be absolute or
      relative to ``CMAKE_CURRENT_SOURCE_DIR``.
    :param FILES:  Files containing messages, relative to `msgdir`
-   :param PACKAGE:  Optional alternate packagename (if the current project doesn't match the
+   :param BASE_DIR:  Optional alternate packagename (if the current project doesn't match the
```
Comment (dirk-thomas) [inline doc/usermacros.rst]: Can you please also:
- update the signature in line 4 as well as
- the description of the `BASE_DIR` argument in this line

Comment (vrabaud) [inline doc/usermacros.rst]: what should I put for BASE_DIR ? Not sure that it behaves the same as PACKAGE

Comment (dirk-thomas) [inline doc/usermacros.rst]: The user should probably never use it. It is only used by actions internally: https://github.com/ros/genmsg/blob/indigo-devel/cmake/genmsg-extras.cmake.em#L90-L93

Comment (vrabaud) [inline doc/usermacros.rst]: Ok . so I should remove it ? The PACKAGE option was useful no ? It allowed
messages in a foo project to actually create headers as if they were part
of a foo_msgs package (or anything else).

**Comment by vrabaud** [comment]
@dirk-thomas , How about now ?

File: doc/usermacros.rst [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+.. cmake:macro:: add_message_files(DIRECTORY dir FILES file1 [file2...] [BASE_DIR base_dir] [NOINSTALL])
 
    :param DIRECTORY: Directory containing messages.  May be absolute or
-     relative to ``CMAKE_CURRENT_SOURCE_DIR``.
+     relative to ``CMAKE_CURRENT_SOURCE_DIR``.  The name will also be used as the install folder. Default is `msg`
    :param FILES:  Files containing messages, relative to `msgdir`
-   :param PACKAGE:  Optional alternate packagename (if the current project doesn't match the
-     desired namespace for the messages)
+   :param BASE_DIR:  Absolute path used and only used as an alternate source message folder.  Has
+     precedence over DIRECTORY.
```
Comment (dirk-thomas) [inline doc/usermacros.rst]: Maybe add:

```
Only used by `add_action_files()`.
```

File: doc/usermacros.rst [hunk 3]
```diff
@@ -1,13 +1,13 @@
 User macro reference
 ====================
 
-.. cmake:macro:: add_message_files(DIRECTORY dir FILES file1 [file2...] [PACKAGE pkgname] [NOINSTALL])
+.. cmake:macro:: add_message_files(DIRECTORY dir FILES file1 [file2...] [BASE_DIR base_dir] [NOINSTALL])
 
    :param DIRECTORY: Directory containing messages.  May be absolute or
-     relative to ``CMAKE_CURRENT_SOURCE_DIR``.
+     relative to ``CMAKE_CURRENT_SOURCE_DIR``.  The name will also be used as the install folder. Default is `msg`
```
Comment (dirk-thomas) [inline doc/usermacros.rst]: If you want to mention the install location it should be the "full" path: `share/<pkgname>/<directory>`

**Comment by vrabaud** [comment]
updated by fixing your comments

**Comment by dirk-thomas** [comment]
Thank you for the patch and iterating on it.

**Pull request merged**